core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -307,10 +307,16 @@ fun validatePathfindingResponse(
         }
     }
 
-    val tracks = HashSet<String>()
-    for (track in res.trackSectionRanges) tracks.add(track.trackSection)
-    if (tracks.size != res.trackSectionRanges.size)
-        throw OSRDError(ErrorType.PathWithRepeatedTracks)
+    val dirTrackDuplicates =
+        res.trackSectionRanges.filter { trackSectionRange ->
+            res.trackSectionRanges
+                .filter {
+                    trackSectionRange.trackSection == it.trackSection &&
+                        trackSectionRange.direction == it.direction
+                }
+                .size > 1
+        }
+    if (dirTrackDuplicates.isNotEmpty()) throw OSRDError(ErrorType.PathWithRepeatedTracks)
```

Comment:
That would scale quite badly when there's a lot of track sections.

We could instead do it the same way as before (map to a set and check if the size change), but mapping to a `Pair(trackId, direction)`